omnibus/package-scripts/iot-agent/postinst

```diff
@@ -14,79 +16,91 @@ DISTRIBUTION=$(lsb_release -d 2>/dev/null | grep -Eo $KNOWN_DISTRIBUTION || grep
-			usermod -g dd-agent dd-agent
-		fi
+    if [ "$DISTRIBUTION_FAMILY" = "Debian" ]; then
+        set -e
+        case "$1" in
+            configure)
+                # Only create dd-agent group and/or user if they don't already exist
+                getent group dd-agent >/dev/null || (echo "Creating dd-agent group" && addgroup --system dd-agent --quiet)
+                set +e
+                id -u dd-agent >/dev/null 2>&1
+                USER_EXISTS=$?
+                set -e
+                if [ ! $USER_EXISTS -eq 0 ]; then
+                    echo "Creating dd-agent user"
+                    adduser --system dd-agent --disabled-login --shell /usr/sbin/nologin --home ${INSTALL_DIR} --no-create-home --group --quiet
+                elif id -nG dd-agent | grep --invert-match --word-regexp --quiet 'dd-agent'; then
+                    # User exists but is not part of the dd-agent group
+                    echo "Adding dd-agent user to dd-agent group"
+                    usermod -g dd-agent dd-agent
+                fi
 
-		# Create a symlink to the agent's binary
-		ln -sf $INSTALL_DIR/bin/agent/agent /usr/bin/datadog-agent
-		;;
-	abort-upgrade | abort-remove | abort-deconfigure) ;;
+                # Create a symlink to the agent's binary
+                ln -sf $INSTALL_DIR/bin/agent/agent /usr/bin/datadog-agent
+            ;;
+            abort-upgrade|abort-remove|abort-deconfigure)
+            ;;
 
-	\
-		*) ;;
+            *)
+            ;;
+        esac
+        #DEBHELPER#
+    fi
 
-	esac
-	#DEBHELPER#
+    # Set proper rights to the dd-agent user
+    chown -R dd-agent:dd-agent ${CONFIG_DIR}
+    chown -R dd-agent:dd-agent ${LOG_DIR}
+    chown -R dd-agent:dd-agent ${INSTALL_DIR}
 
-	# Set proper rights to the dd-agent user
-	chown -R dd-agent:dd-agent ${CONFIG_DIR}
-	chown -R dd-agent:dd-agent ${LOG_DIR}
-	chown -R dd-agent:dd-agent ${INSTALL_DIR}
+    # Enable and restart the agent service here on Debian platforms
+    # On RHEL, this is done in the posttrans script
+    if [ "$DISTRIBUTION_FAMILY" = "Debian" ]; then
+        # Only supports systemd and upstart
+        echo "Enabling service $SERVICE_NAME"
+        if command -v systemctl >/dev/null 2>&1; then
+            # Force systemd to ignore the sysvinit scripts. Only cosmetic, remove some irrelevant warnings during upgrade
+            SYSTEMCTL_SKIP_SYSV=true systemctl enable $SERVICE_NAME || echo "[ WARNING ]\tCannot enable $SERVICE_NAME with systemctl"
+        elif command -v initctl >/dev/null 2>&1; then
+            # Nothing to do, this is defined directly in the upstart job file
+            :
+        elif command -v update-rc.d >/dev/null 2>&1; then
+            update-rc.d $SERVICE_NAME defaults || echo "[ WARNING ]\tCannot enable $SERVICE_NAME with update-rc.d"
+        else
+            echo "[ WARNING ]\tCannot detect a supported init system. The datadog-agent package only provides service files for systemd, upstart and sysvinit."
```

Comment:
```suggestion
            echo "[ WARNING ]\tCannot detect a supported init system. The datadog-iot-agent package only provides service files for systemd, upstart and sysvinit."
```

Reply:
Are we really supporting `sysvinit`? I only see `upstart` & `systemd` mentioned in the project file: https://github.com/DataDog/datadog-agent/blob/29fbbea6d6dc44dcd4f347331cedbf48743977f1/omnibus/config/projects/iot-agent.rb#L76-L95